### PR TITLE
Fixed reference Refinements

### PR DESCRIPTION
**Pull request by waseem-chamaa**
changed refinement to give id on get- calls for References and objects of find-calls

### CODE REVIEW

File: src/foam/nanos/auth/CachedUserAndGroupAuthService.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public Boolean checkPermission(foam.core.X x, java.security.Permission permissio
     User user = (User) x.get("user");
     if ( user == null ) return false;
 
-    Group group = (Group) user.getGroup();
+    Group group =  user.findGroup(x);
```
Comment (jlhughes) [inline src/foam/nanos/auth/CachedUserAndGroupAuthService.java]: fix spacing.

File: src/foam/dao/JDAOJava.js [hunk 2]
```diff
@@ -96,9 +96,12 @@ foam.CLASS({
       name: 'put_',
       synchronized: true,
       javaCode: `
-        if ( ! ( obj instanceof LastModifiedByAware ) || ((LastModifiedByAware) obj).getLastModifiedBy() == null ) {
+      try {
+        if (!(obj instanceof LastModifiedByAware) || ((LastModifiedByAware) obj).findLastModifiedBy(x) == null) {
```
Comment (jlhughes) [inline src/foam/dao/JDAOJava.js]: fix spacing: if (

File: src/foam/dao/JDAOJava.js [hunk 3]
```diff
@@ -107,9 +110,12 @@ foam.CLASS({
       name: 'remove_',
       synchronized: true,
       javaCode: `
-        if ( ! ( obj instanceof LastModifiedByAware ) || ((LastModifiedByAware) obj).getLastModifiedBy() == null ) {
+      try {
+        if (!(obj instanceof LastModifiedByAware) || ((LastModifiedByAware) obj).findLastModifiedBy(x) == null) {
```
Comment (jlhughes) [inline src/foam/dao/JDAOJava.js]: fix spacing: if ( !

File: src/foam/dao/JDAOJava.js [hunk 4]
```diff
@@ -96,9 +96,12 @@ foam.CLASS({
       name: 'put_',
       synchronized: true,
       javaCode: `
-        if ( ! ( obj instanceof LastModifiedByAware ) || ((LastModifiedByAware) obj).getLastModifiedBy() == null ) {
+      try {
+        if (!(obj instanceof LastModifiedByAware) || ((LastModifiedByAware) obj).findLastModifiedBy(x) == null) {
           writeComment((foam.nanos.auth.User) x.get("user"));
         }
+      } catch (Throwable t) {
+        t.printStackTrace();}
```
Comment (kgrgreer) [inline src/foam/dao/JDAOJava.js]: Fix spacing, add a TODO: since this shouldn't be happening.

File: src/foam/dao/JDAOJava.js [hunk 5]
```diff
@@ -107,9 +110,12 @@ foam.CLASS({
       name: 'remove_',
       synchronized: true,
       javaCode: `
-        if ( ! ( obj instanceof LastModifiedByAware ) || ((LastModifiedByAware) obj).getLastModifiedBy() == null ) {
+      try {
+        if (!(obj instanceof LastModifiedByAware) || ((LastModifiedByAware) obj).findLastModifiedBy(x) == null) {
           writeComment((foam.nanos.auth.User) x.get("user"));
         }
+      } catch (Throwable t) {
+        t.printStackTrace();}
```
Comment (kgrgreer) [inline src/foam/dao/JDAOJava.js]: Fix spacing add TODO.

File: src/foam/java/refinements.js [hunk 6]
```diff
@@ -1419,38 +1419,42 @@ foam.CLASS({
...
   ],
 
   methods: [
     function buildJavaClass(cls) {
-      this.SUPER(cls);
+      // Disable super behaviour on purpose.
+      // this.SUPER(cls);
+
+      // Install a renamed copy of the refernced model's id property instead
+      this.referencedProperty.buildJavaClass(cls);
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:1445]: Great Job! This is brilliant!

**Review by jlhughes — APPROVED** [review]

**Pull request merged**